### PR TITLE
27885 refactor i18n select, add breadcrumb lang attr

### PR DESCRIPTION
**Pull request by adamwhitlock1**
## Description
One last accessibility issue was discovered for translated pages, which was that the breadcrumb at the top of the page did not have the lang attribute set, and this PR fixes that issue.

Minor refactoring was also done to centralize constants, move non-hook code to a helpers file, and generally clean up the code a bit to reduce code duplication.

Regression testing was done on a local environment for the translated pages, and also checked on the review instance of this PR.

Sample url for review instance:
[http://6344e9cc749b06b3431d6daed5686595.review.vetsgov-internal/health-care/covid-19-vaccine-esp/](http://6344e9cc749b06b3431d6daed5686595.review.vetsgov-internal/health-care/covid-19-vaccine-esp/)

## Original issue(s)
department-of-veterans-affairs/va.gov-team#27619

## Testing done
Regression testing to verify lang attributes are present on main content, links, and breadcrumbs.
Unit testing for core functions.

## Screenshots
Screen Shot 2021-08-05 at 11 41 30 AM

## Acceptance criteria
- [x] As a user the breadcrumb for the current translated page should have the appropriate `lang` attribute set
- [x] On a translated page, the main content element's `lang` attribute should be set correctly
- [x] All links on a translated page should have the appropriate `hreflang` and `lang` attribute set

## Definition of done
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

**Review by theodur — APPROVED** [review]

**Review by Rleahy22 — COMMENTED** [review]
A few non-blocking comments 🚀

**Comment by Rleahy22** [comment]
Annnnnnd I see I'm a little late. Sorry about that.

File: src/applications/static-pages/i18Select/utilities/helpers.js [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+      const { code } = getConfigFromUrl(link.href, TRANSLATED_LANGUAGES);
+
+      // sidebar links are considered to be translated to their target language,
+      // and therefore would have their hreflang and lang set to their parsed language
+      if (code !== 'en') {
+        link.setAttribute('hreflang', code);
+        link.setAttribute('lang', code);
+      }
+    }
+  }
```
[the remark this answers is not in the document]
Comment (adamwhitlock1) [reply src/applications/static-pages/i18Select/utilities/helpers.js:107]: @Rleahy22 Those are all good points! I was going to look at adding some better test coverage, so I'll keep these in mind for that PR. Thanks!

File: src/applications/static-pages/i18Select/tests/parseLangCode.unit.spec.js [hunk 2]
```diff
@@ -1,42 +1,49 @@
...
     for (const path of paths) {
-      expect(parseLangCode(path)).to.equal('tl');
+      expect(getConfigFromUrl(path, TRANSLATED_LANGUAGES)).to.equal(
+        ALL_LANGUAGES[1],
+      );
     }
   });
 
-  it('should return "en" code for non-matching url patterns', () => {
+  it('should return "tl" code for various url patterns', () => {
```
[the remark this answers is not in the document]
Comment (adamwhitlock1) [reply src/applications/static-pages/i18Select/tests/parseLangCode.unit.spec.js:36]: It looks like it changed from 'en' to 'tl' when actually I had wanted to organize the tests a little better and moved the 'en' testing bit to the top followed by 'es' and 'tl' parts. Hope that makes sense

**Pull request merged**